### PR TITLE
Fixes the detectives hat

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Someone removed hat storage a couple years ago and the hat has been critically lacking in its ability to store candycorn.

### CODE REVIEW

File: code/modules/clothing/head/jobs.dm [hunk 1]
```diff
@@ -55,6 +55,20 @@
 	desc = "There's only one man who can sniff out the dirty stench of crime, and he's likely wearing this hat."
 	icon_state = "detective"
 	armor = list(melee = 25, bullet = 5, laser = 25, energy = 10, bomb = 0, bio = 0, rad = 0)
+	var/candy_cooldown = 0
+
+/obj/item/clothing/head/det_hat/AltClick()
+	..()
+	if(ismob(loc))
+		var/mob/M = loc
+		if(candy_cooldown+120 < world.time)
```
Comment (Iamgoofball) [inline code/modules/clothing/head/jobs.dm]: why not just make it able to store a candy corn and start with one in it

Comment (KorPhaeron) [inline code/modules/clothing/head/jobs.dm]: What if he gets hungry and wants another one

Comment (Iamgoofball) [inline code/modules/clothing/head/jobs.dm]: maybe he should go to the chef :^)

Comment (KorPhaeron) [inline code/modules/clothing/head/jobs.dm]: The chef can't actually make candycorn, unless I'm missing something

File: code/modules/clothing/head/jobs.dm [hunk 2]
```diff
@@ -55,6 +55,20 @@
...
+
+/obj/item/clothing/head/det_hat/AltClick()
+	..()
+	if(ismob(loc))
+		var/mob/M = loc
+		if(candy_cooldown+120 < world.time)
+			var/obj/item/weapon/reagent_containers/food/snacks/candy_corn/CC = new /obj/item/weapon/reagent_containers/food/snacks/candy_corn(src)
+			M.put_in_hands(CC)
+			M << "You slip a candy corn from your hat."
+			candy_cooldown = world.time
```
Comment (duncathan) [inline code/modules/clothing/head/jobs.dm]: Make this `candy_cooldown = world.time + 120`. Then make the if up there `if(candy_cooldown < world.time)`.

Comment (duncathan) [inline code/modules/clothing/head/jobs.dm]: Also, are you sure you want it to be +120? Unless I'm mistaken, that's only twelve seconds.

**Comment by MrPerson** [comment]
Shit

**Comment by PKPenguin321** [comment]
finally

**Comment by Dorsisdwarf** [comment]
Ready?

**Comment by KorPhaeron** [comment]
Yeah it's ready whenever

**Comment by KorPhaeron** [comment]
@Jordie0608 @Miauw62

**Comment by Miauw62** [comment]
well i guess i just started reading mspa again so

**Comment by AnonymousNow** [comment]
Good. I hardly play detective, but I remember the pain the old storage pull caused.

**Pull request merged**